### PR TITLE
SDK-2041 Windows Stop calling v1 profile and v1 logout

### PR DESCRIPTION
**Pull request by NidhiDixit09**
## Reference
SDK-2041 -- [Windows] Stop calling /v1/profile and /v1/logout.
https://branch.atlassian.net/browse/SDK-2041

## Summary
Store and clear the identity locally without making requests to the endpoints.

## Type Of Change

- [ ] New feature (non-breaking change which adds functionality)

## Testing Instructions

->Some files have been removed and package versions are updated. So please check code compilation is successful.
-> Following scenarios should work-
 * User should be able to set identity and then retrieve it later.
 * User should get back following  params via callback for setIdentity and logout.
        -- randomized_bundle_token
        -- session_id
        -- link
* User should be able to logout. On logout only identity is cleared.
* SetIdentity and logout should not work if tracking is disabled. 
-> Ensure No network calls are made for endpoints /v1/profile and v1/logout(check in logs).

cc @BranchMetrics/saas-sdk-devs for visibility.

### CODE REVIEW

**Comment by echo-branch** [comment]
I'd like to do a code walk through on this one. The calls are removed, but the handling when called prior to open seems different from other platforms.

**Comment by echo-branch** [comment]
Just to note what we've found. Since the logout is removed, if the server had a value saved it cannot be cleared and will always return to the client. Need to decide if we ignore it, use only if a device side value is not set, or something else. Will need to talk to product about this.

File: BranchSDK/src/BranchIO/Branch.cpp [hunk 1]
```diff
@@ -248,28 +246,48 @@ Branch::setIdentity(const String& userId, IRequestCallback* callback) {
...
-                getSessionInfo().setBundleToken(bundleToken);
-                storage.setString("session.randomized_bundle_token", bundleToken);
-            }
-        });
-        sendEvent(event, callback);
+            return;
+        }
+        
+        IStorage& storage(Storage::instance());
+        storage.remove("session.identity");
```
Comment (NidhiDixit09) [inline BranchSDK/src/BranchIO/Branch.cpp:258]: @echo-branch  Its removing from storage.

Comment (echo-branch) [reply BranchSDK/src/BranchIO/Branch.cpp:258]: Ok. I think I figured out why the behavior was different from expectations. On windows we don't support setting identity prior to install event?

**Review by echo-branch — APPROVED** [review]
I'm making a follow up ticket for setting id prior to init. Just so it's consistent with iOS and Android. Does not need to be done with this ticket though.

**Pull request merged**